lib/sockjs-parser.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2011-2012 VMware, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ *     Status API Training Shop Blog About Pricing
+ */
+
+// Piece of code picked up and adapted from sockjs-client:
+// https://github.com/sockjs/sockjs-client/blob/0b4879d5e0b7f44d9f8accb28759fd965f70ea1f/lib/main.js#L237-L288
+
+exports.parse = function(msg) {
+  var self = this
+    , type = msg.slice(0, 1)
+    , content = msg.slice(1)
+    , payload
+    ;
+
+  // first check for messages that don't need a payload
+  switch (type) {
+    case 'o':
+      return { type: 'open' };
+    case 'h':
+      return { type: 'heartbeat' };
+  }
+
+  if (content) {
+    try {
+      payload = JSON.parse(content);
```

Comment:
I think that's because they want to be compatible with browsers that don't implement JSON (IE <= 8) : https://github.com/sockjs/sockjs-client/#supported-transports-by-browser-html-served-from-http-or-https

Also JSON3 is just an implementation of the JSON API in JavaScript:
https://www.npmjs.com/package/json3

Quoting:
> The functions behave exactly as described in the ECMAScript spec, except for the date serialization discrepancy noted below.